### ISSUE TITLE
Branching - JGE/JNL, JL/JNGE and JLE/JNG failed. 

### ISSUE THREAD

**Issue by DualBrain** [morphx666/x8086NetEmu#5]
I've put together a small program to test branching and found several branch ops fail:

The file I'm attaching has been modified to focus on just the failed ops in order to (hopefully) be able to repro easier.

[branch2.zip](https://github.com/morphx666/x8086NetEmu/files/629325/branch2.zip)

(I found this code in a forum from 2011 while searching (what seems like an impossible search) for any programs that can exercise an x86 emulator.)

I don't know if this is causing the issue(s); however, it's a start.

**Comment by morphx666** [comment]
I've always suspected that the code responsible for setting the flags has issues and this appears to confirm it.
Is the source code for branch2 available?

**Comment by morphx666** [comment]
Just tested the program with fake86 (an 8086 emulator that works perfectly) and branch2 also reports FAILED on all tests... so I'm afraid there's some bug in this test utility.

**Comment by morphx666** [comment]
hmmm... interesting. the Program passes all tests under VirtualPC running MS-DOS 6.22

**Comment by morphx666** [comment]
I've just updated the code with several fixes, including the emulation of the PUSHF, POPF, SAHF and LAHF opcodes. This fix allows the emulator to pass all of branch2.exe tests.
It does not, however, solve any of the per-existing issues with EDIT, QBASIC, DEFRAG, CHECKIT, etc...

**Comment by DualBrain** [comment]
I'm attaching the source and binary for the complete tests that I found.

[testea_and_branch.zip](https://github.com/morphx666/x8086NetEmu/files/629771/testea_and_branch.zip)

For branch2 test; I just adding a jump to the tests that were failing.

Interestingly enough... these are by the author of Fake86 from a forum post. ;-)

I'll continue looking for additional ways to test...

Merged your changes and confirmed that these tests are passing... closing this this issue.
